server/src/test/Parsing.test.ts

```diff
@@ -163,6 +201,51 @@ complete StudyConcept
     method getStudyData(PatientGroupPartition group) function
     --> domains StudyData
 ;
+`,
+            );
+            done();
+        });
+    });
+
+    describe('ymlId rule', () => {
+        it('should accept different kinds of ymlId', (done) => {
+            checkInputValidityForRule((parser) => parser.ymlId(), `abc`);
+            checkInputValidityForRule((parser) => parser.ymlId(), `a:bc`);
+            checkInputValidityForRule((parser) => parser.ymlId(), `a:bc:d`);
+            checkInputValidityForRule((parser) => parser.ymlId(), `a:bc::d`);
+            checkInputValidityForRule((parser) => parser.ymlId(), `a:bc:::d`);
```

Comment:
Would something with strictly more than three colons be accepted? Would it make sense in YML? For the Engine?
If you reject those or end up editing stuff to reject them, consider adding a test for that (provided that there isn’t one already).

Reply:
> Would something with strictly more than three colons be accepted? Would it make sense in YML? For the Engine?

Nothing more should be accepted it doesn't make sense and I guess it would make the compilation fails too.

> If you reject those or end up editing stuff to reject them, consider adding a test for that (provided that there isn’t one already).

I don't have any test proving that an expression that should fail really fails.

Reply:
Is there something preventing you from adding some tests like this?

Reply:
Yes: lack of time.